libs/category/src/reducers/category/category.reducer.ts

```diff
@@ -104,3 +116,96 @@ export function categoryReducer(state = initialState, action: DaffCategoryAction
       return state;
   }
 }
+
+function toggleCategoryFilter(
+	toggledFilter: DaffToggledCategoryFilterRequest, 
+	appliedFilters: DaffCategoryFilterRequest[]
+): DaffCategoryFilterRequest[] {
+	return isFilterApplied(toggledFilter, appliedFilters)
+		? removeFilter(toggledFilter, appliedFilters)
+		: addFilter(toggledFilter, appliedFilters)
+}
+
+function isFilterApplied(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[]): boolean {
+	const filterMatchingName = appliedFilters.find(filter => toggledFilter.name === filter.name);
+	console.log(filterMatchingName)
+	if(!!filterMatchingName && filterMatchingName.type === DaffCategoryFilterType.Match) return filterMatchingName.value === toggledFilter.value;
+	else return !!(<DaffCategoryFilterEqualRequest>filterMatchingName).value.filter(value => value === toggledFilter.value).length;
+}
+
+function removeFilter(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[])
+	: DaffCategoryFilterRequest[] {
+	return toggledFilter.type === DaffCategoryFilterType.Match ?
+		removeMatchFilter(toggledFilter.name, appliedFilters) :
+		removeEqualOrRangeFilter(toggledFilter, appliedFilters);
+}
+
+function removeMatchFilter(filterName: string, appliedFilters: DaffCategoryFilterRequest[]): DaffCategoryFilterRequest[] {
+	return appliedFilters.filter(filter => filter.name !== filterName)
+}
+
+function removeEqualOrRangeFilter(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[]): DaffCategoryFilterRequest[] {
+	return appliedFilters.map(appliedFilter => {
+		if(appliedFilter.name === toggledFilter.name) {
+			appliedFilter.value = (<DaffCategoryFilterEqualRequest>appliedFilter).value.filter(value => value !== toggledFilter.value);
+		}
+		return appliedFilter;
+	}).filter(filter => filter.type === DaffCategoryFilterType.Match || filter.value.length !== 0)
+}
+
+function addFilter(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[])
+	: DaffCategoryFilterRequest[] {
+	return toggledFilter.type === DaffCategoryFilterType.Match ?
+		addMatchFilter(toggledFilter, appliedFilters) :
+		addEqualOrRangeFilter(toggledFilter, appliedFilters);
+}
+
+function addMatchFilter(toggledFilter: DaffToggledCategoryFilterRequest, appliedFilters: DaffCategoryFilterRequest[])
+: DaffCategoryFilterRequest[] {
+	return appliedFilters.concat([{
```

Comment:
This can be simplified to `appliesFilters.concat([toggledFilter])` I think.

Reply:
it can't. typing errors